src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VersionFileConflictResolver.cs

```diff
@@ -0,0 +1,425 @@
+                {
+                    DependencyDetail newerVersion = repoSemVer > vmrSemVer ? repoVersion! : vmrVersion!;
+                    _logger.LogInformation(
+                        "Asset {assetName} updated to {repoVersion} in the repo and {vmrVersion} in the VMR. Choosing the newer version {newerVersion}",
+                        assetName,
+                        repoVersion.Version,
+                        vmrVersion.Version,
+                        newerVersion.Version);
+                    versionUpdates.Add(newerVersion);
+                    continue;
+                }
+
+                // We can't tell which one is newer, we pick the repo version
+                _logger.LogInformation(
+                    "Asset {assetName} updated to {repoVersion} in the repo and {vmrVersion} in the VMR. Choosing the repo version {repoVersion}",
+                    assetName,
+                    repoVersion.Version,
+                    vmrVersion!.Version,
+                    repoVersion.Version);
+                versionUpdates.Add(repoVersion);
+                continue;
+            }
+
+            if (repoRemoval)
+            {
+                if (vmrAddition)
+                {
+                    // Asset was removed from the repo and added to the VMR at the same time
+                    _logger.LogInformation(
+                        "Asset {assetName} was removed from the repo and added to the VMR at the same time. Skipping the asset",
+                        assetName);
+                    throw new ConflictingDependencyUpdateException(repoChange!, vmrChange!);
+                }
+
+                // Asset got removed from the repo, we can skip it (it won't be in the target V.D.xml)
+                _logger.LogInformation("Asset {assetName} was removed from the repo", assetName);
+                continue;
+            }
+
+            if (vmrRemoval)
+            {
+                if (repoAddition)
+                {
+                    // Asset was removed from the VMR and added to the repo at the same time
+                    _logger.LogInformation(
+                        "Asset {assetName} was removed from the VMR and added to the repo at the same time. Skipping the asset",
+                        assetName);
+                    throw new ConflictingDependencyUpdateException(repoChange!, vmrChange!);
+                }
+
+                // Asset got removed from the VMR, we need to remove it
+                removals.Add(assetName);
+                _logger.LogInformation("Asset {assetName} was removed from the VMR, removing from repo too", assetName);
+                continue;
+            }
+
+            if (repoAddition)
+            {
+                // Asset was added to the repo, no change necessary
+                _logger.LogInformation("Asset {assetName} was added in the repo", assetName);
+                additions.Add(repoVersion!);
```

Comment:
When this get's "processed" it won't do anything I think, so we might aswell remove it

Reply:
I found out through tests that it's actually better to "re-add" it just to be sure because when there's a conflict in the version files, when we resolve it (`git checkout --ours`), some dependencies might disappear. So when we have an already ongoing PR branch and are flowing a new update there, it might get ugly.

This is harmless, kinda easy to read and makes sure we end up with the right dependencies anyway.